### PR TITLE
Fix displaying WORK or TRANSACTION after BEGIN

### LINKED ISSUE #1553: `BEGIN;` is formatted as `BEGIN TRANSACTION;`, which is invalid in MySQL
`BEGIN;` is valid in MySQL, but not `BEGIN TRANSACTION;`: https://dev.mysql.com/doc/refman/8.4/en/commit.html

But sqlparser does not preserve the exact syntax of the statement, and always formats it as  `BEGIN TRANSACTION;`, causing an execution error in MySQL.

Found while investigating https://github.com/sqlpage/SQLPage/issues/711

### PR DESCRIPTION
**Pull request by mvzink**
Fixes  #1553

### CODE REVIEW

File: tests/sqlparser_sqlite.rs [hunk 1]
```diff
@@ -527,9 +527,6 @@ fn parse_start_transaction_with_modifier() {
     sqlite_and_generic().verified_stmt("BEGIN DEFERRED TRANSACTION");
     sqlite_and_generic().verified_stmt("BEGIN IMMEDIATE TRANSACTION");
     sqlite_and_generic().verified_stmt("BEGIN EXCLUSIVE TRANSACTION");
-    sqlite_and_generic().one_statement_parses_to("BEGIN DEFERRED", "BEGIN DEFERRED TRANSACTION");
-    sqlite_and_generic().one_statement_parses_to("BEGIN IMMEDIATE", "BEGIN IMMEDIATE TRANSACTION");
-    sqlite_and_generic().one_statement_parses_to("BEGIN EXCLUSIVE", "BEGIN EXCLUSIVE TRANSACTION");
```
Comment (iffyio) [inline tests/sqlparser_sqlite.rs:532]: were the removed tests containing invalid sql?

File: tests/sqlparser_sqlite.rs [hunk 2]
```diff
@@ -527,9 +527,9 @@ fn parse_start_transaction_with_modifier() {
...
+    sqlite_and_generic().verified_stmt("BEGIN DEFERRED");
```
Comment (alamb) [inline tests/sqlparser_sqlite.rs:530]: nice

**Review by alamb — APPROVED** [review]
Thank you @mvzink

**Review by iffyio — APPROVED** [review]
LGTM!

**Comment by alamb** [comment]
🚀

**Pull request merged**